core/domain/story_domain.py

```diff
@@ -285,13 +297,15 @@ def validate(self):
 class StoryContents(object):
     """Domain object representing the story_contents dict."""
 
-    def __init__(self, story_nodes):
+    def __init__(self, story_nodes, starting_node_id):
         """Constructs a StoryContents domain object.
 
         Args:
             story_nodes: list(StoryNode). The list of story nodes that are part
                 of this story.
+            starting_node_id: str. The id of the starting node of the story.
         """
+        self.starting_node_id = starting_node_id
         self.nodes = story_nodes
 
     def validate(self):
```

Comment:
Btw are we validating uniqueness of all node IDs?

If not, might want to double check that we are validating absolutely everything that needs to be validated. Better to overvalidate if possible.

Reply:
Done, added the validation, though node ids would be auto generated, so there is a good chance that they won't duplicate.

Reply:
Hm, we should be certain. Thanks for adding the validation.